devolutions-gateway/src/api/network_scan.rs

```diff
@@ -115,23 +115,49 @@ pub struct NetworkScanResponse {
 }
 
 impl NetworkScanResponse {
-    fn new(ip: IpAddr, port: u16, dns: Option<String>) -> Self {
+    fn new(ip: IpAddr, port: u16, dns: Option<String>, service: Option<scanner::Protocol>) -> Self {
         let hostname = dns;
 
-        let protocol = match port {
-            22 => ApplicationProtocol::Known(Protocol::Ssh),
-            23 => ApplicationProtocol::Known(Protocol::Telnet),
-            80 => ApplicationProtocol::Known(Protocol::Http),
-            443 => ApplicationProtocol::Known(Protocol::Https),
-            389 => ApplicationProtocol::Known(Protocol::Ldap),
-            636 => ApplicationProtocol::Known(Protocol::Ldaps),
-            3389 => ApplicationProtocol::Known(Protocol::Rdp),
-            5900 => ApplicationProtocol::Known(Protocol::Vnc),
-            5985 => ApplicationProtocol::Known(Protocol::WinrmHttpPwsh),
-            5986 => ApplicationProtocol::Known(Protocol::WinrmHttpsPwsh),
-            _ => ApplicationProtocol::unknown(),
+        let protocol = if service.is_some() {
+            network_scanner_protocol_to_gateway_protocol(service)
```

Comment:
suggestion: And retrieve a `scanner::Protocol` using pattern matching.

```suggestion
        let protocol = if let Some(protocol) = service {
            network_scanner_protocol_to_gateway_protocol(protocol)
```